libyul/AsmJsonConverter.cpp

```diff
@@ -170,27 +170,31 @@ Json::Value AsmJsonConverter::operator()(ForLoop const& _node) const
 
 Json::Value AsmJsonConverter::operator()(Break const& _node) const
 {
-	return createAstNode(nativeLocationOf(_node), "YulBreak");
+	return createAstNode(originLocationOf(_node), nativeLocationOf(_node), "YulBreak");
 }
 
 Json::Value AsmJsonConverter::operator()(Continue const& _node) const
 {
-	return createAstNode(nativeLocationOf(_node), "YulContinue");
+	return createAstNode(originLocationOf(_node), nativeLocationOf(_node), "YulContinue");
 }
 
 Json::Value AsmJsonConverter::operator()(Leave const& _node) const
 {
-	return createAstNode(nativeLocationOf(_node), "YulLeave");
+	return createAstNode(originLocationOf(_node), nativeLocationOf(_node), "YulLeave");
 }
 
-Json::Value AsmJsonConverter::createAstNode(langutil::SourceLocation const& _location, string _nodeType) const
+Json::Value AsmJsonConverter::createAstNode(langutil::SourceLocation const& _location, langutil::SourceLocation const& _nativeLocation, string _nodeType) const
 {
 	Json::Value ret{Json::objectValue};
 	ret["nodeType"] = std::move(_nodeType);
 	int length = -1;
 	if (_location.start >= 0 && _location.end >= 0)
 		length = _location.end - _location.start;
 	ret["src"] = to_string(_location.start) + ":" + to_string(length) + ":" + (m_sourceIndex.has_value() ? to_string(m_sourceIndex.value()) : "-1");
+	length = -1;
+	if (_nativeLocation.start >= 0 && _nativeLocation.end >= 0)
+		length = _nativeLocation.end - _nativeLocation.start;
+	ret["nativeSrc"] = to_string(_nativeLocation.start) + ":" + to_string(length) + ":" + (m_sourceIndex.has_value() ? to_string(m_sourceIndex.value()) : "-1");
```

Comment:
Maybe this could be a lambda function, since its use for `src` and `nativeSrc` is almost the same.

Reply:
Something like:
```c++
	auto srcLocation = [&](int start, int end) -> string
	{
		int length = -1;
		if (start >= 0 && end >= 0)
			length = end - start;
		return to_string(start) + ":" + to_string(length) + ":" + (m_sourceIndex.has_value() ? to_string(m_sourceIndex.value()) : "-1");
	};
	ret["src"] = srcLocation(_location.start, _location.end);
	ret["nativeSrc"] = srcLocation(_nativeLocation.start, _nativeLocation.end);

```
And maybe rename `_location` to `originLocation`?